airflow/jobs/job.py

```diff
@@ -104,12 +106,13 @@ class Job(Base, LoggingMixin):
     Only makes sense for SchedulerJob and BackfillJob instances.
     """
 
-    def __init__(self, executor=None, heartrate=None, **kwargs):
+    def __init__(self, executor: BaseExecutor | None = None, heartrate=None, **kwargs):
         # Save init parameters as DB fields
         self.heartbeat_failed = False
         self.hostname = get_hostname()
         if executor:
             self.executor = executor
```

Comment:
Reminder to remove this, either here directly or in a later PR.

Reply:
I'll cut a good-first-issue for this :+1: